src/celeritas/global/alongstep/detail/UrbanMsc.hh

```diff
@@ -101,28 +101,33 @@ CELER_FUNCTION void UrbanMsc::calc_step(CoreTrackView const& track,
     auto particle = track.make_particle_view();
     auto geo      = track.make_geo_view();
     auto phys     = track.make_physics_view();
+    auto sim      = track.make_sim_view();
 
     // Sample multiple scattering step length
     UrbanMscStepLimit msc_step_limit(msc_params_,
                                      particle,
                                      phys,
                                      track.make_material_view().material_id(),
+                                     sim.num_steps() == 0,
                                      geo.find_safety(),
                                      local->step_limit.step);
 
     auto rng             = track.make_rng_engine();
     auto msc_step_result = msc_step_limit(rng);
-    track.make_physics_step_view().msc_step(msc_step_result);
+    track.make_physics_step_view().msc_step(msc_step_result.msc_step);
+
+    // Store range properties for multiple scattering
+    phys.msc_range(msc_step_result.msc_range);
```

Comment:
I thought you'd only want to do this if the range isn't calculated yet? Instead of returning this by value, maybe you should take the PhysicsTrackView as a non-const input to the MSC step length, and store the step if it's the first step or entering a new volume.

Reply:
Changed the input as suggested and updated codes accordingly.